flashcards/backend/decks/models.py

```diff
@@ -20,3 +20,33 @@ def __str__(self):
     class Meta:
         verbose_name = "category"
         verbose_name_plural = "categories"
+
+
+class DifficultyLevel(models.Model):
+    """Model for representing how hard flashcard/deck is to learn
+
+    :param name: level name
+    :type name: str
+    :param value: value
+    :type value: int
+
+    """
+
+    name = models.CharField(max_length=20, unique=True)
+    value = models.PositiveSmallIntegerField()
+
```

Comment:
@MWasile I was thinking about it. I have this idea that setting `unique` only for name isn't enough, because we might end up with two different names with the same `value`.  Meaning `DifficultyLevel(name="SomeName", value=10)` and `DifficultyLevel(name="SomeOtherName", value=10)`, which I find a bit weird. So my suggestion is to add a class Meta option with `unique_together`

https://docs.djangoproject.com/en/4.1/ref/models/options/#unique-together

Reply:
@MWasile @martynawitkowska whilst we are at this should I also go with adding constrains with Lower to Meta as recomended to @BlazejBielski by Martyna in [this branch ](https://github.com/AzenoIT/azeno/pull/44/files) instead of overriding save method?

Reply:
In the branch, you are talking about is a bit different story, because tags, in general, are lowercased, but I don't think for difficulty level it is good.

Reply:
So after some thinking, I believie that you could remove the custom save method. Add unique constraint with `Lower` option and `unique_together`

Reply:
I have replaced custom save with Unique Constrain, however adding unique together did not work as I though (ie. I could add several difficulty levels with value 1) . So instead i made value unique which ensured that we have only one difficulty  level with given value, I hope this is what was expected. Let me know what you think?

Reply:
Nicely done 😊

Reply:
> do we still use typing?

@bartoszbaran2 not sure, but i was under impression that model fields are already typed. Let me know about your thought on that matter.